src/inference/src/dev/core_impl.cpp

```diff
@@ -54,8 +54,251 @@ void stripDeviceName(std::string& device, const std::string& substr) {
     }
 }
 
+bool is_virtual_device(const std::string& device_name) {
+    return (device_name.find("AUTO") != std::string::npos || device_name.find("MULTI") != std::string::npos ||
+            device_name.find("HETERO") != std::string::npos || device_name.find("BATCH") != std::string::npos);
+};
+
+ov::AnyMap clone_map(const ov::AnyMap& m) {
+    ov::AnyMap rm;
+    for (auto&& kvp : m) {
+        rm[kvp.first] = kvp.second.is<ov::AnyMap>() ? ov::Any(clone_map(kvp.second.as<ov::AnyMap>())) : kvp.second;
+    }
```

Comment:
why have to do recursive clone here

Reply:
removed
added tests that Any does not share the values